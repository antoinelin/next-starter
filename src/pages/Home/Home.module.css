.container {
  @apply min-h-screen px-2 flex flex-col justify-center items-center h-screen;
}

.main {
  @apply py-20 flex flex-1 flex-col justify-center items-center;
}

.title a {
  @apply text-blue-600 no-underline;
}

.title a:hover,
.title a:focus,
.title a:active {
  @apply underline;
}

.title {
  @apply m-0 leading-normal text-6xl;
}

.title,
.description {
  @apply text-center;
}

.description {
  @apply text-2xl;
}

.code {
  @apply bg-gray-200 rounded p-3 text-lg font-mono;
}

.grid {
  @apply flex flex-col items-center justify-center flex-wrap max-w-3xl w-full mt-12;
}

@screen md {
  .grid {
    @apply flex-row;
  }
}
