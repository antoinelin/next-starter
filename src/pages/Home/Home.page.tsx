import Card from '@/components/Card/Card.component'
import Head from 'next/head'
import styles from './Home.module.css'

export default function Home() {
  return (
    <div className={styles.container}>
      <Head>
        <title>Create Next App</title>
        <meta name="description" content="Generated by create next app" />
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <main className={styles.main}>
        <h1 className={styles.title}>
          Welcome to <a href="https://nextjs.org">Next.js!</a>
        </h1>

        <p className={styles.description}>
          Get started by editing <code className={styles.code}>pages/index.js</code>
        </p>

        <div className={styles.grid}>
          <Card
            title="Documentation"
            description="Find in-depth information about Next.js features and API."
            link="https://nextjs.org/docs"
          />
          <Card
            title="Learn"
            description="Learn about Next.js in an interactive course with quizzes!"
            link="https://nextjs.org/learn"
          />
          <Card
            title="Examples"
            description="Discover and deploy boilerplate example Next.js projects."
            link="https://nextjs.org/examples"
          />
          <Card
            title="Deploy"
            description="Instantly deploy your Next.js site to a public URL with Vercel."
            link="https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
          />
        </div>
      </main>
    </div>
  )
}

Home.displayName = 'Home'
