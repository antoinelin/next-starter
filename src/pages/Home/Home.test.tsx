import { render, screen } from '@testing-library/react'
import { getPage } from 'next-page-tester'
import React from 'react'
import renderer from 'react-test-renderer'
import Home from './Home.page'

describe('Home', () => {
  test('renders Home page', async () => {
    const { render } = await getPage({
      route: '/',
    })

    render()
    expect(screen.getByText('Next.js!')).toBeInTheDocument()
  })

  test('renders a heading', () => {
    render(<Home />)

    const heading = screen.getByRole('heading', {
      name: /welcome to next\.js!/i,
    })

    expect(heading).toBeInTheDocument()
  })

  test('renders homepage unchanged', () => {
    const tree = renderer.create(<Home />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
