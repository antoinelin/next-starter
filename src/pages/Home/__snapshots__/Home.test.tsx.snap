// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots renders homepage unchanged 1`] = `
<div
  className="container"
>
  <main
    className="main"
  >
    <h1
      className="title"
    >
      Welcome to 
      <a
        href="https://nextjs.org"
      >
        Next.js!
      </a>
    </h1>
    <p
      className="description"
    >
      Get started by editing 
      <code
        className="code"
      >
        pages/index.js
      </code>
    </p>
    <div
      className="grid"
    >
      <a
        className="card"
        href="https://nextjs.org/docs"
      >
        <h2>
          Documentation
           →
        </h2>
        <p>
          Find in-depth information about Next.js features and API.
        </p>
        <svg
          className="logo"
        >
          <use
            xlinkHref="/icons.svg#vercel"
          />
        </svg>
      </a>
      <a
        className="card"
        href="https://nextjs.org/learn"
      >
        <h2>
          Learn
           →
        </h2>
        <p>
          Learn about Next.js in an interactive course with quizzes!
        </p>
        <svg
          className="logo"
        >
          <use
            xlinkHref="/icons.svg#vercel"
          />
        </svg>
      </a>
      <a
        className="card"
        href="https://nextjs.org/examples"
      >
        <h2>
          Examples
           →
        </h2>
        <p>
          Discover and deploy boilerplate example Next.js projects.
        </p>
        <svg
          className="logo"
        >
          <use
            xlinkHref="/icons.svg#vercel"
          />
        </svg>
      </a>
      <a
        className="card"
        href="https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
      >
        <h2>
          Deploy
           →
        </h2>
        <p>
          Instantly deploy your Next.js site to a public URL with Vercel.
        </p>
        <svg
          className="logo"
        >
          <use
            xlinkHref="/icons.svg#vercel"
          />
        </svg>
      </a>
    </div>
  </main>
</div>
`;
